src/S3/PostObjectV4.php

```diff
@@ -0,0 +1,191 @@
+<?php
+namespace Aws\S3;
+
+use Aws\Credentials\CredentialsInterface;
+use GuzzleHttp\Psr7\Uri;
+use Aws\Signature\SignatureTrait;
+use Aws\Common\Enum\DateFormat;
+use Aws\Signature\SignatureV4 as SignatureV4;
+use Aws\Api\TimestampShape as TimestampShape;
+
+/**
+ * Encapsulates the logic for getting the data for an S3 object POST upload form
+ *
+ * @link http://docs.aws.amazon.com/AmazonS3/latest/API/RESTObjectPOST.html
+ * @link http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-post-example.html
+ */
+class PostObjectV4
+{
+    use SignatureTrait;
+
+    private $client;
+    private $bucket;
+    private $formAttributes;
+    private $formInputs;
+    private $jsonPolicy;
+
+    /**
+     * Constructs the PostObject.
+     *
+     * The options array accepts the following keys:
+     * @link http://docs.aws.amazon.com/AmazonS3/latest/API/sigv4-query-string-auth.html
+     *
+     * @param S3ClientInterface $client     Client used with the POST object
+     * @param string            $bucket     Bucket to use
+     * @param array             $formInputs Associative array of form input
+     *                                      fields.
+     * @param array             $options    Policy condition options
+     * @param mixed             $expiration Upload expiration time value. By
+     *                                      default: 1 hour vaild peroid.
+     */
+    public function __construct(
+        S3ClientInterface $client,
+        $bucket,
+        array $formInputs,
+        array $options = [],
+        $expiration = '+1 hours'
+    ) {
+        $this->client = $client;
+        $this->bucket = $bucket;
+
+        // setup form attributes
+        $this->formAttributes = [
+            'action'  => $this->generateUri(),
+            'method'  => 'POST',
+            'enctype' => 'multipart/form-data'
+        ];
+
+        // setup basic policy
+        $expiration = TimestampShape::format($expiration, 'iso8601');
+
+        $policy = [
+            'expiration' => $expiration,
+            'conditions' => [
+                ['X-Amz-Algorithm' => "AWS4-HMAC-SHA256"],
```

Comment:
This is being added in `getPolicyAndSignature`, isn't it? `$policy` here can just be defined as:
```php
$policy = [
    'expiration' => TimestampShape::format($expiration, 'iso8601'),
    'conditions' => $options,
];
```

That would capture your intention here more concisely by communicating the role of `$options` here.